encoding/json/decode.go

```diff
@@ -908,13 +909,26 @@ func (d *Decoder) decodeFieldType(valueJSON any, results typeDecodingResults) ca
 	)
 }
 
-func (d *Decoder) decodeFunctionType(returnValue, parametersValue, id any, results typeDecodingResults) cadence.Type {
+func (d *Decoder) decodePurity(purity any) cadence.FunctionPurity {
+	functionPurity := toString(purity)
+	if functionPurity == "pure" {
+		return cadence.PureFunction
+	} else if functionPurity == "impure" {
+		return cadence.ImpureFunction
+	} else {
+		panic(ErrInvalidJSONCadence)
+	}
```

Comment:
Can we make "purity" optional in the JSON, so that users only have to add it if it is pure (less breaking)